common/src/main/java/dev/architectury/hooks/item/FoodPropertiesHooks.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * This file is part of architectury.
+ * Copyright (C) 2020, 2021 architectury
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package dev.architectury.hooks.item;
+
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.food.FoodProperties;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Supplier;
+
+public final class FoodPropertiesHooks {
+    private FoodPropertiesHooks() {
+    }
+    
+    public static void effect(@NotNull FoodProperties.Builder builder,
+                              @NotNull Supplier<? extends MobEffectInstance> effectSupplier, float chance) {
+        ((FoodPropertiesBuilderAccess) builder).effect(effectSupplier, chance);
```

Comment:
Would @ExpectPlatform here, to resolve the get() for the supplier on fabric immediately, and pass the supplier to forge.
We don't need the deferring mob effect on fabric.